server/src-lib/Hasura/RQL/DML/Select/Types.hs

```diff
@@ -142,19 +142,34 @@ data ColOp
   , _colExp :: S.SQLExp
   } deriving (Show, Eq)
 
+data AnnColField
+  = AnnColField
+  { _acfInfo   :: !PGColumnInfo
+  , _acfAsText :: !Bool
```

Comment:
The context in the PR description is super helpful, and it would be nice if some of it could make it into the code, since otherwise it’s kind of hard to tell what this field is for if you’re just reading the code. Could you maybe add a short blurb about what this is for, with a note like “see #3198 for more details” at the end? Just to keep everything tied together.